app/helpers/user_auth.rb

```diff
@@ -0,0 +1,17 @@
+def current_user
+  @current_user ||= User.find_by(id: session[:user_id])
+end
+
+def logged_in?
+  !!current_user
+end
+
+def owns_suggestion(suggestion)
+  logged_in? && suggestion.user == current_user
```

Comment:
Looks like you have the method that you could use to protect against unauthorized access.